design-docs/trajectories.md

```diff
@@ -0,0 +1,188 @@
+# Trajectories 2: Better Edition
+
+## Problem 
+
+WPILib currently has a very limited trajectory API;
+it only allows for cubic or quintic splines,
+and only for tangential heading interpolation.
+This works for differential drive robots,
+but holonomic robots are able to follow more complex trajectories.
+
+This document describes a new trajectory API that can be used
+by both differential drive and holonomic robots,
+and a builder-style trajectory generator that can generate trajectories
+for both types of robot.
+
+## Goals 
+
+1. Define a trajectory API that can represent a wide variety of trajectories,
+   including those generated by Choreo and PathPlanner.
+2. Create a trajectory generator that can generate trajectories
+   for both differential drive and holonomic robots.
+3. Allow future trajectory generation algorithms to be added
+   without changing the API.
+
+## Non-Goals
+
+1. Replace tools like Choreo and PathPlanner. 
+   The new Trajectory API is intended to be used by those tools,
+   not to replace them.
+   We do not expect the external libraries to use our generation API,
+   but they should be able to generate trajectories that extend our `Trajectory` class.
+2. Implement a full trajectory following system.
+   This document only describes the trajectory API and generator.
+
+## Design: Trajectory API
+
+External libraries such as Choreo and PathPlanner should be able to
+generate trajectories that extend the `Trajectory` class presented here,
+to allow for users to add external trajectory generation algorithms
+without changing their trajectory *following* code.
+
+There are a few new classes that this API will use:
+
+### ChassisAcceleration 
+
+The `ChassisAcceleration` class represents the acceleration of a chassis.
+It is basically just ChassisSpeeds except it is for accelerations instead of velocities:
+
+```java
+public class ChassisAccelerations {
```

Comment:
If we add this, we'll need to make it work with the kinematics class somehow, since forward and inverse kinematics should work the same way with it.